tests/test_transferlearning.py

```diff
@@ -0,0 +1,246 @@
+        labels_target = get_labels(n_matrices_target, n_classes)
+
+        trans = transfer(transfer_coef=0.5, metric={mean_keys: 'riemann',
+                         dist_keys: 'riemann'})
+        trans.fit(covmats_target, labels_target, covmats_source,
+                  labels_source, sample_weight)
+
+
+@pytest.mark.parametrize("transfer", [MDWM])
+def test_metric_wrong_type(transfer, get_covmats, get_labels):
+    with pytest.raises((TypeError)):
+        n_classes, n_channels = 2, 3
+        n_matrices_class_source, n_matrices_class_target = 3, 2
+        n_matrices_source = n_classes * n_matrices_class_source
+        n_matrices_target = n_classes * n_matrices_class_target
+        sample_weight = None
+
+        covmats_source = get_covmats(n_matrices_source, n_channels)
+        labels_source = get_labels(n_matrices_source, n_classes)
+
+        covmats_target = get_covmats(n_matrices_target, n_channels)
+        labels_target = get_labels(n_matrices_target, n_classes)
+
+        trans = transfer(transfer_coef=0.5, metric=12)
+        trans.fit(covmats_target, labels_target, covmats_source,
+                  labels_source, sample_weight)
+
+
+@pytest.mark.parametrize("transfer", [MDWM])
+@pytest.mark.parametrize("transfer_coef", [-0.1, 1.1])
+def test_transfer_coef_range(transfer, transfer_coef, get_covmats, get_labels):
+    with pytest.raises((ValueError)):
+        n_classes, n_channels = 2, 3
+        n_matrices_class_source, n_matrices_class_target = 3, 2
+        n_matrices_source = n_classes * n_matrices_class_source
+        n_matrices_target = n_classes * n_matrices_class_target
+        sample_weight = None
+
+        covmats_source = get_covmats(n_matrices_source, n_channels)
+        labels_source = get_labels(n_matrices_source, n_classes)
+
+        covmats_target = get_covmats(n_matrices_target, n_channels)
+        labels_target = get_labels(n_matrices_target, n_classes)
+
+        trans = transfer(transfer_coef=transfer_coef)
+        trans.fit(covmats_target, labels_target, covmats_source,
+                  labels_source, sample_weight)
+
+
+@pytest.mark.parametrize("transfer", [MDWM])
+def test_different_labels_error(transfer, get_covmats, get_labels):
+    with pytest.raises((ValueError)):
+        n_classes, n_classes_source, n_channels = 6, 4, 3
+        n_matrices_class_source, n_matrices_class_target = 3, 2
+        n_matrices_source = n_classes_source * n_matrices_class_source
+        n_matrices_target = n_classes * n_matrices_class_target
+        sample_weight = None
+
+        covmats_source = get_covmats(n_matrices_source, n_channels)
+        labels_source = get_labels(n_matrices_source, n_classes_source)
+
+        covmats_target = get_covmats(n_matrices_target, n_channels)
+        labels_target = get_labels(n_matrices_target, n_classes)
+
+        trans = transfer(transfer_coef=0.5)
+        trans.fit(covmats_target, labels_target, covmats_source,
+                  labels_source, sample_weight)
```

Comment:
there is a lot of code duplication between the tests

Reply:
yes, with source/target data, transfer learning makes lengthy tests, but each test use fixture and specific variable names to make the object of the test more explicit. With @emmanuelkalunga we tried to factorize the tests as much as possible.